torch_np/_binary_ufuncs.py

```diff
@@ -77,69 +77,57 @@ def matmul(
 
     # NB: do not broadcast input tensors against the out=... array
     result = _binary_ufuncs.matmul(*tensors)
-    return _helpers.result_or_out(result, out)
-
-
-#
-# For each torch ufunc implementation, decorate and attach the decorated name
-# to this module. Its contents is then exported to the public namespace in __init__.py
-#
-for name in __all__:
-    ufunc = getattr(_binary_ufuncs, name)
-    decorated = normalizer(deco_binary_ufunc(ufunc))
-
-    decorated.__qualname__ = name  # XXX: is this really correct?
-    decorated.__name__ = name
-    vars()[name] = decorated
-
-
-# a stub implementation of divmod, should be improved after
-# https://github.com/pytorch/pytorch/issues/90820 is fixed in pytorch
-#
-# Implementation details: we just call two ufuncs which have been created
-# just above, for x1 // x2 and x1 % x2.
-# This means we are normalizing x1, x2 in each of the ufuncs --- note that there
-# is no @normalizer on divmod.
+    return result
 
 
 def divmod(
-    x1,
-    x2,
+    x1: ArrayLike,
+    x2: ArrayLike,
+    out1: Optional[NDArray] = None,
+    out2: Optional[NDArray] = None,
     /,
-    out=None,
+    out: Optional[tuple[NDArray]] = (None, None),
```

Comment:
nit `tuple[Optional[NDArray], Optional[NDArray]]`.

Reply:
Why? It's either `(None, None)` or `(NDArray, NDArray)`; can't have `(out1, None)` (I _think_)

Reply:
`tuple`, as opposed, to `list` takes the number of arguments that it accepts explicitly. As such, there you are declaring there a tuple of length 1 and one NDArray. 
Then, `Optional[tuple[NDArray, NDArray]]` accepts either a tuple of ndarrays (fine) or `None` (not `(None, None)`).
What you want is to have a (partial) function that accepts `tuple[Optional[NDArray], Optional[NDArray]]` and errors out when just one of the elements of the tuple is `None`.

Reply:
Feel free to patch this small correction in any other PR.

Reply:
Ouch. python typing meta-language is seriously weird.

Reply:
Note that this is not specific to Python, the same would be true in C++. This would be an `std::tuple<std::optional<array, std::optional<array>>` rather than a `std::optional<std::tuple<array, array>>` :)